### PR TITLE
feat: Add public/private IPs & block volume attachments

### LINKED ISSUE #10573: Oracle Source Plugin: Add VNIC IPs and Block Volume Attachments
Hi, I wanted to say thank you for creating this plugin!  It will definitely make keeping track of assets in our Oracle tenancy much easier.
In looking through the postgres tables I created as my destination for the oci data, I am having trouble finding vnic ips  and block volume attachments to instances to tie those assets together.

Is there a plan to add additional surface area to this plugin as the above would really be helpful information for managing these cloud assets?  I'd certainly be interested in helping to implement though as a Java developer primarily, golang is relatively new to me.  Below are a couple of references I found for the aforementioned SDK functionality.

[ListVolumeAttachments](https://docs.oracle.com/en-us/iaas/api/#/en/iaas/20160918/VolumeAttachment/ListVolumeAttachments) 

```
https://github.com/oracle/oci-java-sdk/blob/master/bmc-examples/src/main/java/GetInstancePublicIpExample.java

     Iterable<PrivateIp> privateIpsIterable =
                    vcnClient
                            .getPaginators()
                            .listPrivateIpsRecordIterator(
                                    ListPrivateIpsRequest.builder().vnicId(vnicId).build());
            for (PrivateIp privateIp : privateIpsIterable) {
                try {
                    GetPublicIpByPrivateIpIdResponse getPublicIpResponse =
                            vcnClient.getPublicIpByPrivateIpId(
                                    GetPublicIpByPrivateIpIdRequest.builder()
                                            .getPublicIpByPrivateIpIdDetails(
                                                    GetPublicIpByPrivateIpIdDetails.builder()
                                                            .privateIpId(privateIp.getId())
                                                            .build())
                                            .build());
                    publicIps.add(getPublicIpResponse.getPublicIp().getIpAddress());
                } catch (BmcException e) {
                    // A 404 is expected if the private IP address does not have a public IP
                    if (e.getStatusCode() != 404) {
                        System.out.println(
                                String.format(
                                        "Exception when retriving public IP for private IP %s (%s)",
                                        privateIp.getId(), privateIp.getIpAddress()));
                    } else {
                        System.out.println(
                                String.format(
                                        "No public IP for private IP %s (%s)",
                                        privateIp.getId(), privateIp.getIpAddress()));
                    }
                }
            }
        }
```
Regards,
Chris Whelan

### PR DESCRIPTION
**Pull request by candiduslynx**
Closes #10573

### CODE REVIEW

File: plugins/source/oracle/resources/services/compute/volume_attachments.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+	"github.com/cloudquery/cloudquery/plugins/source/oracle/client"
+	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v3/transformers"
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/core"
+)
+
+type VolumeAttachment struct {
+	// Both core.ParavirtualizedVolumeAttachment & core.EmulatedVolumeAttachment
```
Comment (erezrokah) [inline plugins/source/oracle/resources/services/compute/volume_attachments.go:14]: Based on the docs it seems `EmulatedVolumeAttachment` is not supported?
See https://pkg.go.dev/github.com/oracle/oci-go-sdk/core#ComputeClient.ListVolumeAttachments
image

Not sure we can do anything about it but maybe update the comment?

**Review by erezrokah — COMMENTED** [review]
Thanks for the PR @candiduslynx, added a few comments

File: plugins/source/oracle/resources/services/compute/volume_attachments.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/oracle/resources/services/compute/volume_attachments.go:14]: The one missing has just the same fields, so I can either update the comment or leave as is

Comment (candiduslynx) [reply plugins/source/oracle/resources/services/compute/volume_attachments.go]: Simplified in c3b6fc0da36df4ba2a39a3da5b4d62bf48264cd5

File: plugins/source/oracle/resources/services/compute/volume_attachments.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	// This will be filled in and used only if the underlying attachment is core.IScsiVolumeAttachment.
+	// It gives us extra fields to report, too.
+	*core.IScsiVolumeAttachment
+}
+
+func VolumeAttachments() *schema.Table {
+	return &schema.Table{
+		Name:                "oracle_compute_volume_attachments",
+		Resolver:            fetchVolumeAttachments,
+		PreResourceResolver: unwrapVolumeAttachment,
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply plugins/source/oracle/resources/services/compute/volume_attachments.go]: simplified in c3b6fc0da36df4ba2a39a3da5b4d62bf48264cd5

**Review by erezrokah — APPROVED** [review]

**Pull request merged**